### PR TITLE
Cache function symbol resoltuion when calling from SIDE_MODULE into M…

### PR DESCRIPTION
**Pull request by sbc100**
…AIN_MODULE

Without this change function calls from the side module
into the main module call resolveSymbol each time they
are run.

### CODE REVIEW

File: src/library_dylink.js [hunk 1]
```diff
@@ -304,8 +304,10 @@ var LibraryDylink = {
             };
           }
           // otherwise this is regular function import - call it indirectly
+          var resolved;
           return obj[prop] = function() {
-            return resolveSymbol(prop, 'function', true).apply(null, arguments);
+            if (!resolved) resolved = resolveSymbol(prop, 'function', true);
```
Comment (kripken) [inline src/library_dylink.js:309]: can this be outside of the `function()`, 2 lines earlier? or is it resolved on the actual first call? a comment might help if so.

Comment (sbc100) [reply src/library_dylink.js:309]: No the idea is that resolution is not done until the function is first called.  If we try to resolve now most of symbol would be missing.

**Comment by sbc100** [comment]
OK to land?

**Comment by sbc100** [comment]
ping

**Review by kripken — APPROVED** [review]

**Pull request merged**